### PR TITLE
Adding geometric information on DisplayList items while dumping.

### PR DESCRIPTION
**Pull request by Adenilson**

### CODE REVIEW

**Comment by pcwalton** [comment]
Can you eliminate the code duplication by implementing `Display` on `Rect<Au>` (or at least on a newtype struct of it; e.g. `struct AuRect(Rect<Au>); impl Display for AuRect { ... }`

**Comment by Adenilson** [comment]
Sure, will do it.

**Comment by Adenilson** [comment]
Not for commit as it will need some companion changes on rust-geom.

**Comment by Adenilson** [comment]
Squashed and updated the deps.

File: components/gfx/display_list/mod.rs [hunk 1]
```diff
@@ -169,15 +169,13 @@ impl DisplayList {
...
-                    DisplayItem::GradientClass(ref _gradient) => println!("{} GradientClass.", indentation),
-                    DisplayItem::LineClass(ref _line) => println!("{} LineClass.", indentation),
-                    DisplayItem::BoxShadowClass(ref _box_shadow) => println!("{} BoxShadowClass.", indentation),
+                    DisplayItem::SolidColorClass(ref solid_color) => println!("{} SolidColor. {}", indentation, solid_color.base.bounds),
+                    DisplayItem::TextClass(ref text) => println!("{} Text. {}", indentation, text.base.bounds),
+                    DisplayItem::ImageClass(ref image) => println!("{} Image. {}", indentation, image.base.bounds),
+                    DisplayItem::BorderClass(ref border) => println!("{} Border. {}", indentation, border.base.bounds),
+                    DisplayItem::GradientClass(ref gradient) => println!("{} Gradient. {}", indentation, gradient.base.bounds),
+                    DisplayItem::LineClass(ref line) => println!("{} Line. {}", indentation, line.base.bounds),
+                    DisplayItem::BoxShadowClass(ref box_shadow) => println!("{} Box_shadow. {}", indentation, box_shadow.base.bounds),
```
Comment (bjz) [inline components/gfx/display_list/mod.rs]: Couldn't you use `DisplayItem::LineClass(ref line) => println!("{:?} Line. {:?}", indentation, line.base.bounds),`?

**Comment by Adenilson** [comment]
yeah, it works. Let me push a patch on top of it.

**Comment by Adenilson** [comment]
I feel that all remarks made by reviewers have being addressed, should I squash and push it?

File: components/util/geometry.rs [hunk 2]
```diff
@@ -84,6 +84,15 @@ impl Zero for Au {
...
 }
 
+impl fmt::Display for Au {
+    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
+        let Au(app_unit) = self.clone();
+        let tmp = app_unit.to_string();
+        try!(formatter.write_str(tmp.as_slice()));
+        Ok(())
+    }
+}
```
Comment (bjz) [inline components/util/geometry.rs]: Don't need this I don't think?

Comment (Adenilson) [inline components/util/geometry.rs]: @bjz when I first tried to implement this, the compiler barked that there was no way to print a Au and it was missing the Show/String ops.

**Comment by Adenilson** [comment]
I see. {:?} will use the Debug ops.

**Comment by bjz** [comment]
LGTM. Please squash.

**Comment by Adenilson** [comment]
Squashed and pushed.

**Comment by Adenilson** [comment]
Just rebased.

**Comment by Adenilson** [comment]
OOps, there is left over (the Formatter). Let me upload an update patch without it.

**Comment by Adenilson** [comment]
Done.

**Pull request closed without merging**